internal/incident/incident.go

```diff
@@ -60,351 +73,475 @@ func (i *Incident) HasManager() bool {
 	return false
 }
 
-func (i *Incident) AddHistory(historyRow *HistoryRow, fetchId bool) (types.Int, error) {
-	historyRow.IncidentID = i.incidentRowID
+// ProcessEvent processes the given event for the current incident.
+func (i *Incident) ProcessEvent(ev event.Event, created bool) error {
+	i.Lock()
+	i.runtimeConfig.RLock()
 
-	stmt := utils.BuildInsertStmtWithout(i.db, historyRow, "id")
-	if fetchId {
-		historyId, err := utils.InsertAndFetchId(i.db, stmt, historyRow)
-		if err != nil {
-			return types.Int{}, err
-		}
+	defer func() {
+		i.Unlock()
+		i.runtimeConfig.RUnlock()
+	}()
```

Comment:
Just use two separate `defer` statements, you don't gain much from putting this in a function. Also, I'd put them right after the corresponding lock statement, because that gives you the teardown order you typically want (though that doesn't really matter here).